abce/agent.py

```diff
@@ -48,6 +48,112 @@
 import traceback
 from abce.notenoughgoods import NotEnoughGoods
 import datetime
+from abce.trade import get_epsilon
+epsilon = get_epsilon()
+
+
+class Inventory(defaultdict):
+    def __init__(self, name):
+        super(Inventory, self).__init__(float)
+        self.name = name
+        self.contracts = set()
+        self._expiring_goods = []
+        self._perishable = []
+        #self.goods = defaultdict(float)
+
+    def create(self, good, quantity):
+        """ creates quantity of the good out of nothing
+
+        Use with care. As long as you use it only for labor and
+        natural resources your model is macro-economically complete.
+
+        Args:
+            'good': is the name of the good
+            quantity: number
+        """
+        self[good] += quantity
+
+    def destroy(self, good, quantity=None):
+        """ destroys quantity of the good. If quantity is omitted destroys all
+
+        Use with care.
+
+        Args::
+
+            'good':
+                is the name of the good
+            quantity (optional):
+                number
+
+        Raises::
+
+            NotEnoughGoods: when goods are insufficient
+        """
+        if quantity is None:
+            self[good] = 0
+        else:
+            available = self[good]
+            if available < quantity - epsilon:
+                self[good] = 0
```

Comment:
This is the current implementation of `destroy`, see https://github.com/ABC-E/abce/blob/651639152519ce5a8d8e6c07c13f6dc44c05bfcf/abce/agent.py#L315.